src/libraries/Common/tests/System/Security/Cryptography/AlgorithmImplementations/MLDsa/MLDsaTestsBase.cs

```diff
@@ -0,0 +1,223 @@
+        {
+            byte[] secretKey = new byte[MLDsaAlgorithm.MLDsa44.SecretKeySizeInBytes];
+            byte[] seed = new byte[MLDsaAlgorithm.MLDsa44.PrivateSeedSizeInBytes];
+            using (MLDsa mldsaOriginal = GenerateKey(MLDsaAlgorithm.MLDsa44))
+            {
+                Assert.Equal(secretKey.Length, mldsaOriginal.ExportMLDsaSecretKey(secretKey));
+                Assert.Equal(seed.Length, mldsaOriginal.ExportMLDsaPrivateSeed(seed));
+            }
+
+            using (MLDsa mldsa = ImportPrivateSeed(MLDsaAlgorithm.MLDsa44, seed))
+            {
+                byte[] secretKey2 = new byte[MLDsaAlgorithm.MLDsa44.SecretKeySizeInBytes];
+                byte[] seed2 = new byte[MLDsaAlgorithm.MLDsa44.PrivateSeedSizeInBytes];
+
+                Assert.Equal(secretKey2.Length, mldsa.ExportMLDsaSecretKey(secretKey2));
+                Assert.Equal(seed2.Length, mldsa.ExportMLDsaPrivateSeed(seed2));
+
+                AssertExtensions.SequenceEqual(secretKey, secretKey2);
+                AssertExtensions.SequenceEqual(seed, seed2);
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(MLDsaTestsData.AllNistTestCases), MemberType = typeof(MLDsaTestsData))]
+        public void NistImportPublicKeyVerify(MLDsaNistTestCase testCase)
+        {
+            using MLDsa mldsa = ImportPublicKey(testCase.Algorithm, testCase.PublicKey);
+            Assert.Equal(testCase.ShouldPass, mldsa.VerifyData(testCase.Message, testCase.Signature, testCase.Context));
+        }
+
+        [Theory]
+        [MemberData(nameof(MLDsaTestsData.AllNistTestCases), MemberType = typeof(MLDsaTestsData))]
+        public void NistImportSecretKeyVerifyExportsAndSignature(MLDsaNistTestCase testCase)
+        {
+            using MLDsa mldsa = ImportSecretKey(testCase.Algorithm, testCase.SecretKey);
+
+            byte[] pubKey = new byte[testCase.Algorithm.PublicKeySizeInBytes];
+            Assert.Equal(pubKey.Length, mldsa.ExportMLDsaPublicKey(pubKey));
+            AssertExtensions.SequenceEqual(testCase.PublicKey, pubKey);
+
+            byte[] secretKey = new byte[testCase.Algorithm.SecretKeySizeInBytes];
+            Assert.Equal(secretKey.Length, mldsa.ExportMLDsaSecretKey(secretKey));
+
+            byte[] seed = new byte[testCase.Algorithm.PrivateSeedSizeInBytes];
+            Assert.Throws<CryptographicException>(() => mldsa.ExportMLDsaPrivateSeed(seed));
+
+            Assert.Equal(testCase.ShouldPass, mldsa.VerifyData(testCase.Message, testCase.Signature, testCase.Context));
+        }
+
+        private static void ExerciseSuccessfulVerify(MLDsa mldsa, byte[] data, byte[] signature, byte[] context)
+        {
+            Assert.True(mldsa.VerifyData(data, signature, context));
+            data[0] ^= 1;
+            Assert.False(mldsa.VerifyData(data, signature, context));
+            data[0] ^= 1;
+
+            signature[0] ^= 1;
+            Assert.False(mldsa.VerifyData(data, signature, context));
```

Comment:
It's probably virtuous to switch all of these new True/False asserts to their AssertExtensions.{bool}Expression variant.